apps/wing/src/commands/pack.ts

```diff
@@ -0,0 +1,196 @@
+import { constants } from "fs";
+import * as fs from "fs/promises";
+import * as os from "os";
+import * as path from "path";
+import { resolve } from "path";
+import Arborist from "@npmcli/arborist";
+import { Target } from "@winglang/compiler";
+import { minimatch } from "minimatch";
+import packlist from "npm-packlist";
+import * as tar from "tar";
+import { compile } from "./compile";
+
+// TODO: add --dry-run option?
+// TODO: let the user specify library's supported targets in package.json, and compile to each before packaging
+// TODO: print information about the generated library? (e.g. size, dependencies, number of public APIs)
+
+export interface PackageOptions {
+  /**
+   * Output filename.
+   */
+  readonly outfile?: string;
+}
+
+export async function pack(options: PackageOptions = {}): Promise<string> {
+  // check that the library compiles to the "sim" target
+  console.log('Compiling to the "sim" target...');
+
+  // TODO: workaround for https://github.com/winglang/wing/issues/4431
+  // await compile(".", { target: Target.SIM });
+  await compile(path.join("..", path.basename(process.cwd())), { target: Target.SIM });
+
+  const userDir = process.cwd();
+  const outfile = options.outfile ? resolve(options.outfile) : undefined;
+  const outdir = outfile ? path.dirname(outfile) : userDir;
+
+  // check package.json exists
+  const originalPkgJsonPath = path.join(userDir, "package.json");
+  if (!(await exists(originalPkgJsonPath))) {
+    throw new Error(`No package.json found in the current directory. Run \`npm init\` first.`);
+  }
+
+  const originalPkgJson = JSON.parse(await fs.readFile(originalPkgJsonPath, "utf8"));
+  const originalPkgJsonFiles: Set<string> = new Set(originalPkgJson.files ?? []);
+
+  // perform our work in a staging directory to avoid making a mess in the user's current directory
+  return withTempDir(async (workdir) => {
+    const excludeGlobs = ["/target/**", "/node_modules/**", "/.git/**", "/.wing/**"];
+    const includeGlobs = [
+      ...originalPkgJsonFiles,
+      "README*",
+      "package.json",
+      "**/*.w",
+      "**/*.js",
+      "LICENSE*",
+    ];
+
+    // copy the user's directory to the staging directory
+    await copyDir(userDir, workdir, excludeGlobs, includeGlobs);
+
+    // check package.json exists
+    const pkgJsonPath = path.join(workdir, "package.json");
+    if (!(await exists(pkgJsonPath))) {
+      throw new Error(`No package.json found in the current directory. Run \`npm init\` first.`);
+    }
+
+    const pkgJson = JSON.parse(await fs.readFile(pkgJsonPath, "utf8"));
```

Comment:
Why read the file again? Isn't this the same as the originalJsonFile above?

Reply:
Oops, you're right